Bug Report: Flakiness in Local example

### Overview of the Issue

Ever since VTOrc has been introduced in the local example, the workflow that tests this has been very flaky.

Some of the failed runs - 
- https://github.com/vitessio/vitess/actions/runs/3093773569/jobs/5012933857
- https://github.com/vitessio/vitess/actions/runs/3076563628/jobs/4970806413

### Reproduction Steps

Check the actions and find the failed workflows for `local_example`.

### Binary Version

```sh
main
```


### Operating System and Environment details

```sh
main
```


### Log Fragments

```sh
E0921 10:14:29.095522   15738 main.go:47] rpc error: code = Unknown desc = tablet zone1-0000000402 InitReplica failed: rpc error: code = DeadlineExceeded desc = context deadline exceeded;tablet zone1-0000000401 InitReplica failed: rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

I have spent considerable time investigating this flakiness and I have found 2 issues!

Numero Uno - 
VTOrc runs `ClusterHasNoPrimary` recovery just as it sees the first tablet of a new shard is up. This leads to issues since the durability_policy is semi-sync, so the `PRS` call doesn't succeed and actually blocks on `PopulateReparentJournal` and finally times out. 
Following this, we run `InitShardPrimary` from the local_example script, and that makes things even worse, since `ResetReplication` call on the previously promoted primary also fails. This is because it is stuck on semi-sync ACK and that also blocks `RESET MASTER` calls.

Numero Dos -
VTOrc runs `ClusterHasNoPrimary` and promotes a new primary successfully. The primary starts initializing the backend schema in vttablets, including `schema_migrations` tables which have a bunch of alter queries.
Before all of these queries succeed, we run `InitShardPrimary` which resets replication on all the tablets. This doesn't stop the writes on the primary. We then take the primary's replication position and paste them on the replica's without replicating anything. Then when we try to start replicaiton, it fails complaining 
```
2022-09-21T12:19:45.388093Z 46 [ERROR] [MY-010584] [Repl] Slave SQL for channel '': Worker 1 failed executing transaction 'a4f5dc7c-39a7-11ed-b071-3ed77ce818bb:7' at master log vt-0000000400-bin.000001, end_log_pos 3281; Error 'Table '_vt.schema_migrations' doesn't exist' on query. Default database: ''. Query: 'ALTER TABLE _vt.schema_migrations add column retries int unsigned NOT NULL DEFAULT 0', Error_code: MY-001146
2022-09-21T12:19:45.388247Z 45 [ERROR] [MY-010586] [Repl] Error running query, slave SQL thread aborted. Fix the problem, and restart the slave SQL thread with "SLAVE START". We stopped at log 'vt-0000000400-bin.000001' position 3037
```
This causes replication to be permanently broken and `ISP` to fail.

--- a/examples/local/101_initial_cluster.sh
+++ b/examples/local/101_initial_cluster.sh
@@ -46,14 +46,14 @@ vtctldclient --server localhost:15999 SetKeyspaceDurabilityPolicy --durability-p
 ./scripts/vtorc-up.sh
 
 # Wait for all the tablets to be up and registered in the topology server
-for i in $(seq 0 200); do
+for _ in $(seq 0 200); do
 	vtctldclient GetTablets --keyspace commerce --shard 0 | wc -l | grep -q "3" && break
 	sleep 1
 done;
 vtctldclient GetTablets --keyspace commerce --shard 0 | wc -l | grep -q "3" || (echo "Timed out waiting for tablets to be up in commerce/0" && exit 1)
 
 # Wait for a primary tablet to be elected in the shard
-for i in $(seq 0 200); do
+for _ in $(seq 0 200); do
 	vtctldclient GetTablets --keyspace commerce --shard 0 | grep -q "primary" && break
 	sleep 1
 done;

--- a/examples/local/101_initial_cluster.sh
+++ b/examples/local/101_initial_cluster.sh
@@ -42,8 +42,22 @@ done
 # set the correct durability policy for the keyspace
 vtctldclient --server localhost:15999 SetKeyspaceDurabilityPolicy --durability-policy=semi_sync commerce
 
-# set one of the replicas to primary
-vtctldclient PlannedReparentShard commerce/0 --new-primary zone1-100
+# start vtorc
+./scripts/vtorc-up.sh
+
+# Wait for all the tablets to be up and registered in the topology server
+for i in $(seq 0 200); do
+	vtctldclient GetTablets --keyspace commerce --shard 0 | wc -l | grep -q "3" && break
+	sleep 1
+done;
+vtctldclient GetTablets --keyspace commerce --shard 0 | wc -l | grep -q "3" || (echo "Timed out waiting for tablets to be up in commerce/0" && exit 1)
+
+# Wait for a primary tablet to be elected in the shard
+for i in $(seq 0 200); do
+	vtctldclient GetTablets --keyspace commerce --shard 0 | grep -q "primary" && break
+	sleep 1
+done;
+vtctldclient GetTablets --keyspace commerce --shard 0 | grep "primary" || (echo "Timed out waiting for primary to be elected in commerce/0" && exit 1)
 
 # create the schema
 vtctldclient ApplySchema --sql-file create_commerce_schema.sql commerce
@@ -57,5 +71,3 @@ CELL=zone1 ./scripts/vtgate-up.sh
 # start vtadmin
 ./scripts/vtadmin-up.sh
 
-# start vtorc
-./scripts/vtorc-up.sh

--- a/examples/local/201_customer_tablets.sh
+++ b/examples/local/201_customer_tablets.sh
@@ -29,14 +29,14 @@ done
 vtctldclient --server localhost:15999 SetKeyspaceDurabilityPolicy --durability-policy=semi_sync customer
 
 # Wait for all the tablets to be up and registered in the topology server
-for i in $(seq 0 200); do
+for _ in $(seq 0 200); do
 	vtctldclient GetTablets --keyspace customer --shard 0 | wc -l | grep -q "3" && break
 	sleep 1
 done;
 vtctldclient GetTablets --keyspace customer --shard 0 | wc -l | grep -q "3" || (echo "Timed out waiting for tablets to be up in customer/0" && exit 1)
 
 # Wait for a primary tablet to be elected in the shard
-for i in $(seq 0 200); do
+for _ in $(seq 0 200); do
 	vtctldclient GetTablets --keyspace customer --shard 0 | grep -q "primary" && break
 	sleep 1
 done;

--- a/examples/local/201_customer_tablets.sh
+++ b/examples/local/201_customer_tablets.sh
@@ -28,4 +28,16 @@ done
 # set the correct durability policy for the keyspace
 vtctldclient --server localhost:15999 SetKeyspaceDurabilityPolicy --durability-policy=semi_sync customer
 
-vtctldclient InitShardPrimary --force customer/0 zone1-200
+# Wait for all the tablets to be up and registered in the topology server
+for i in $(seq 0 200); do
+	vtctldclient GetTablets --keyspace customer --shard 0 | wc -l | grep -q "3" && break
+	sleep 1
+done;
+vtctldclient GetTablets --keyspace customer --shard 0 | wc -l | grep -q "3" || (echo "Timed out waiting for tablets to be up in customer/0" && exit 1)
+
+# Wait for a primary tablet to be elected in the shard
+for i in $(seq 0 200); do
+	vtctldclient GetTablets --keyspace customer --shard 0 | grep -q "primary" && break
+	sleep 1
+done;
+vtctldclient GetTablets --keyspace customer --shard 0 | grep "primary" || (echo "Timed out waiting for primary to be elected in customer/0" && exit 1)

--- a/examples/local/201_newkeyspace_tablets.sh
+++ b/examples/local/201_newkeyspace_tablets.sh
@@ -33,14 +33,14 @@ for i in ${BASETABLETNUM}00 ${BASETABLETNUM}01 ${BASETABLETNUM}02; do
 done
 
 # Wait for all the tablets to be up and registered in the topology server
-for i in $(seq 0 200); do
+for _ in $(seq 0 200); do
 	vtctldclient GetTablets --keyspace "${KEYSPACE}" --shard "${SHARD}" | wc -l | grep -q "3" && break
 	sleep 1
 done;
 vtctldclient GetTablets --keyspace "${KEYSPACE}" --shard "${SHARD}" | wc -l | grep -q "3" || (echo "Timed out waiting for tablets to be up in ${KEYSPACE}/${SHARD}" && exit 1)
 
 # Wait for a primary tablet to be elected in the shard
-for i in $(seq 0 200); do
+for _ in $(seq 0 200); do
 	vtctldclient GetTablets --keyspace "${KEYSPACE}" --shard "${SHARD}" | grep -q "primary" && break
 	sleep 1
 done;

--- a/examples/local/201_newkeyspace_tablets.sh
+++ b/examples/local/201_newkeyspace_tablets.sh
@@ -32,7 +32,19 @@ for i in ${BASETABLETNUM}00 ${BASETABLETNUM}01 ${BASETABLETNUM}02; do
         SHARD=${SHARD} CELL=zone1 KEYSPACE=${KEYSPACE} TABLET_UID=$i ./scripts/vttablet-up.sh
 done
 
-vtctldclient InitShardPrimary --force "${KEYSPACE}/${SHARD}" "zone1-${BASETABLETNUM}00"
+# Wait for all the tablets to be up and registered in the topology server
+for i in $(seq 0 200); do
+	vtctldclient GetTablets --keyspace "${KEYSPACE}" --shard "${SHARD}" | wc -l | grep -q "3" && break
+	sleep 1
+done;
+vtctldclient GetTablets --keyspace "${KEYSPACE}" --shard "${SHARD}" | wc -l | grep -q "3" || (echo "Timed out waiting for tablets to be up in ${KEYSPACE}/${SHARD}" && exit 1)
+
+# Wait for a primary tablet to be elected in the shard
+for i in $(seq 0 200); do
+	vtctldclient GetTablets --keyspace "${KEYSPACE}" --shard "${SHARD}" | grep -q "primary" && break
+	sleep 1
+done;
+vtctldclient GetTablets --keyspace "${KEYSPACE}" --shard "${SHARD}" | grep "primary" || (echo "Timed out waiting for primary to be elected in ${KEYSPACE}/${SHARD}" && exit 1)
 
 # Go back to the original ${PWD} in the parent shell
 popd >/dev/null

--- a/examples/local/302_new_shards.sh
+++ b/examples/local/302_new_shards.sh
@@ -29,5 +29,18 @@ for i in 400 401 402; do
 	SHARD=80- CELL=zone1 KEYSPACE=customer TABLET_UID=$i ./scripts/vttablet-up.sh
 done
 
-vtctldclient InitShardPrimary --force customer/-80 zone1-300
-vtctldclient InitShardPrimary --force customer/80- zone1-400
+for shard in "-80" "80-"; do
+  # Wait for all the tablets to be up and registered in the topology server
+  for i in $(seq 0 200); do
+  	vtctldclient GetTablets --keyspace customer --shard $shard | wc -l | grep -q "3" && break
+  	sleep 1
+  done;
+  vtctldclient GetTablets --keyspace customer --shard $shard | wc -l | grep -q "3" || (echo "Timed out waiting for tablets to be up in customer/$shard" && exit 1)
+
+  # Wait for a primary tablet to be elected in the shard
+  for i in $(seq 0 200); do
+  	vtctldclient GetTablets --keyspace customer --shard $shard | grep -q "primary" && break
+  	sleep 1
+  done;
+  vtctldclient GetTablets --keyspace customer --shard $shard | grep "primary" || (echo "Timed out waiting for primary to be elected in customer/$shard" && exit 1)
+done;

--- a/examples/local/302_new_shards.sh
+++ b/examples/local/302_new_shards.sh
@@ -31,14 +31,14 @@ done
 
 for shard in "-80" "80-"; do
   # Wait for all the tablets to be up and registered in the topology server
-  for i in $(seq 0 200); do
+  for _ in $(seq 0 200); do
   	vtctldclient GetTablets --keyspace customer --shard $shard | wc -l | grep -q "3" && break
   	sleep 1
   done;
   vtctldclient GetTablets --keyspace customer --shard $shard | wc -l | grep -q "3" || (echo "Timed out waiting for tablets to be up in customer/$shard" && exit 1)
 
   # Wait for a primary tablet to be elected in the shard
-  for i in $(seq 0 200); do
+  for _ in $(seq 0 200); do
   	vtctldclient GetTablets --keyspace customer --shard $shard | grep -q "primary" && break
   	sleep 1
   done;

--- a/examples/region_sharding/101_initial_cluster.sh
+++ b/examples/region_sharding/101_initial_cluster.sh
@@ -38,7 +38,15 @@ SHARD=0 CELL=zone1 KEYSPACE=main TABLET_UID=100 ./scripts/vttablet-up.sh
 # set the correct durability policy for the keyspace
 vtctldclient --server localhost:15999 SetKeyspaceDurabilityPolicy --durability-policy=none main
 
-vtctldclient InitShardPrimary --force main/0 zone1-100
+# start vtorc
+./scripts/vtorc-up.sh
+
+# Wait for a primary tablet to be elected in the shard
+for _ in $(seq 0 200); do
+	vtctldclient GetTablets --keyspace main --shard 0 | grep -q "primary" && break
+	sleep 1
+done;
+vtctldclient GetTablets --keyspace main --shard 0 | grep "primary" || (echo "Timed out waiting for primary to be elected in main/0" && exit 1)
 
 # create the schema
 vtctldclient ApplySchema --sql-file create_main_schema.sql main
@@ -52,5 +60,3 @@ CELL=zone1 ./scripts/vtgate-up.sh
 # start vtadmin
 ./scripts/vtadmin-up.sh
 
-# start vtorc
-./scripts/vtorc-up.sh

--- a/examples/region_sharding/202_new_tablets.sh
+++ b/examples/region_sharding/202_new_tablets.sh
@@ -26,8 +26,11 @@ SHARD=80-c0 CELL=zone1 KEYSPACE=main TABLET_UID=400 ./scripts/vttablet-up.sh
 CELL=zone1 TABLET_UID=500 ./scripts/mysqlctl-up.sh
 SHARD=c0- CELL=zone1 KEYSPACE=main TABLET_UID=500 ./scripts/vttablet-up.sh
 
-# set primary
-vtctldclient InitShardPrimary --force main/-40 zone1-200
-vtctldclient InitShardPrimary --force main/40-80 zone1-300
-vtctldclient InitShardPrimary --force main/80-c0 zone1-400
-vtctldclient InitShardPrimary --force main/c0- zone1-500
+for shard in "-40" "40-80" "80-c0" "c0-"; do
+  # Wait for a primary tablet to be elected in the shard
+  for _ in $(seq 0 200); do
+  	vtctldclient GetTablets --keyspace main --shard $shard | grep -q "primary" && break
+  	sleep 1
+  done;
+  vtctldclient GetTablets --keyspace main --shard $shard | grep "primary" || (echo "Timed out waiting for primary to be elected in main/$shard" && exit 1)
+done;

